src/pkm/PK3RR.ts

```diff
@@ -0,0 +1,355 @@
+      } else {
+        this.ball = Ball.Poke
+      }
+      this.ivs = other.ivs ?? {
+        hp: 0,
+        atk: 0,
+        def: 0,
+        spe: 0,
+        spa: 0,
+        spd: 0,
+      }
+      this.isEgg = other.isEgg ?? false
+      this.isNicknamed = other.isNicknamed ?? false
+      this.currentHP = other.currentHP
+      this.nickname = other.nickname
+      this.trainerName = other.trainerName
+      this.trainerGender = other.trainerGender
+    }
+  }
+
+  static fromBytes(buffer: ArrayBufferLike): PK3RR {
+    return new PK3RR(buffer)
+  }
+
+  toBytes(options?: types.ToBytesOptions): ArrayBuffer {
+    const buffer = new ArrayBuffer(58); // 58 bytes as specified
+    const dataView = new DataView(buffer);
+
+    // 0:4 Personality
+    dataView.setUint32(0x0, this.personalityValue, true);
+
+    // 4:8 OT ID (Trainer ID and Secret ID)
+    dataView.setUint16(0x4, this.trainerID, true);
+    dataView.setUint16(0x6, this.secretID, true);
+
+    // 8:18 Nickname (10 bytes)
+    stringLogic.writeGen3StringToBytes(dataView, this.nickname, 0x8, 10, false);
+
+    // 18 Language
+    dataView.setUint8(0x12, this.languageIndex);
+
+    // 19 Sanity
+    // dataView.setUint8(0x13, SANITY VALUE IDK);
+
+    // 20:27 OT Name (7 bytes)
+    stringLogic.writeGen3StringToBytes(dataView, this.trainerName, 0x14, 7, true);
+
+    // 27 Markings
+    types.markingsFourShapesToBytes(dataView, 0x1b, this.markings);
+
+    // Growth Substructure (starts at 0x1C)
+    // 28:30 Species (DexNum)
+    dataView.setUint16(0x1c, conversion.toGen3PokemonIndex(this.dexNum), true);
+
+    // 30:32 Held Item
+    dataView.setUint16(0x1e, this.heldItemIndex, true);
+
+    // 32:36 Experience
+    dataView.setUint32(0x20, this.exp, true);
+
+    // 36 PP Bonuses (PP Ups)
+    dataView.setUint8(0x24, this.movePPUps.reduce((acc, ppUp, i) => acc | (ppUp << (i * 2)), 0));
+
+    // 37 Friendship
+    dataView.setUint8(0x25, this.trainerFriendship);
+
+    // 38 Ball (Pokeball)
+    dataView.setUint8(0x26, this.ball);
+
+    // Moves (5 bytes total for 10-bit moves)
+    byteLogic.uIntToBufferBits(dataView, conversion.toGen3RRMoveIndex(this.moves[0]), 0x27, 0, 10, true);
+    byteLogic.uIntToBufferBits(dataView, conversion.toGen3RRMoveIndex(this.moves[1]), 0x28, 2, 10, true);
+    byteLogic.uIntToBufferBits(dataView, conversion.toGen3RRMoveIndex(this.moves[2]), 0x29, 4, 10, true);
+    byteLogic.uIntToBufferBits(dataView, conversion.toGen3RRMoveIndex(this.moves[3]), 0x2A, 6, 10, true);
+
+    // EVs
+    types.writeStatsToBytes(dataView, 0x2b, this.evs);
+
+    // 49 Pokerus
+    dataView.setUint8(0x31, this.pokerusByte);
+
+    // 50 Met Location
+    dataView.setUint8(0x32, this.metLocationIndex);
+
+    // 51:52 Met Info (packed: Met level, Game of Origin, Trainer Gender)
+    byteLogic.uIntToBufferBits(dataView, this.metLevel, 0x33, 0, 7, true);
+    byteLogic.uIntToBufferBits(dataView, this.gameOfOrigin == 6 ? 8 : this.gameOfOrigin, 0x33, 7, 4, true);
```

Comment:
Can you explain what the logic here is doing? 8 is the Game of Origin value for SoulSilver so I'm confused as to why we would want to set it to that. Does Radical Red have its own Game of Origin value?

Reply:
Yeah I believe RR sets the offset to 8 (sometimes, othertimes its other values. I'm not sure what the defining logic is here). But this shifts it down to 6 if the game of origin is 8. 

Also I got the program to build. Had to remove a lot of the unused parameters though. However it not correctly reading each pokemon and there's a lot of issues with the `util.getMaxPP`.

Reply:
Don't mean to hijack this thread but when I open a RR save it says its a HGSS save. Do you know why this is. My thought process has to do with how the saves are differetiated in util.GetSaveType

Reply:
Created a discussion for this.

Reply:
Yes. It builds. But uhh, now it doesn't seem to be reading the Saves.